docs/docs/guides/04-fe-coding-guidelines.mdx

```diff
@@ -0,0 +1,596 @@
+---
+title: FE Coding Guidelines
+---
+
+## Coding Style
+
+### Application Structure
+
+The FE of Strapi aims to be a domain driven application such as we heavily utilise the concept of
+nested routes:
+
+```shell
+pages/
+├─ index.tsx
+├─ settings/
+│  ├─ pages/
+│  │  ├─ index.tsx
+│  │  ├─ users/
+│  │  ├─ roles/
+│  │  │  ├─ create.tsx
+│  │  │  ├─ :id.tsx
+│  │  │  ├─ index.tsx
+```
+
+As part of this effort, assuming a component is not used anywhere else it should live as close as
+possible to the section of the app it's used in:
+
+```shell
+pages/
+├─ index.tsx
+├─ settings/
+│  ├─ components/
+│  │  ├─ SettingsNavigation.tsx
+│  ├─ hooks/
+│  │  ├─ useSettings.ts
+│  ├─ pages/
+│  │  ├─ index.tsx
+│  │  ├─ users/
+│  │  ├─ roles/
+│  │  │  ├─ create.tsx
+│  │  │  ├─ :id.tsx
+│  │  │  ├─ index.tsx
+```
+
+When components/hooks etc. are required higher in the tree, we can subsequently move them higher up
+as necessary. But we strive for colocation.
+
+### Exports
+
+Prefer named exports over default exports. Why? Because Strapi is a _very_ large codebase. If you
+export something as `default` someone can import it with a different name:
+
+```jsx
+// button.tsx
+
+const MyButton = (props) => <button {...props} />;
+
+export default MyButton;
+
+// form.tsx
+import Button from './button';
+
+// NAMED EXPORT
+export const Form = () => <Button type="button">Submit</Button>;
+```
+
+Above is a simple component for illustration purposes, but you'd be likely to use the `MyButton`
+component in _a lot_ of places, so what happens if you want to replace all instances with a new v2
+Button? If the naming is not as consistent, it's typically a lot more frustrating & can lead to
+bugs.
+
+### Utilities / Helpers
+
+When you need a utility function ask yourself a question – do I need this _anywhere_ else? If you
+don't (which is fairly typical) include it in the file you're using it. As a guide, we expect
+anything exported to be unit-tested and if a utility or hook is unique and bespoke to the component
+you're creating then it's advised to test that code in the context of said component, this is
+covered in the Testing guides below.
+
+If you do need to use the utility elsewhere then follow the same practices shown in
+[Application Structure](#application-structure) keep it as close to where it's used as possible.
+Avoid re-exporting them from a generic `index` file. However, you might find it useful to group them
+under a generic file name such as `strings.ts` which would include all the `string` utils you have.
+
```

Comment:
Should we add here that we want to stop using re-exports from `index.ts` files and similar to lodash start removing them subsequently?

Reply:
why can't we make a combination of the two things, the one you mentioned in [this comment](https://github.com/strapi/strapi/pull/17739/files#r1301163952) and re-exporting:

```shell
hooks
├─ tests
├── useContentManager.test.ts
├── useOtherHook.test.ts
├─ useContentManager.ts
├─ useOtherHook.ts
├─ index.ts
```

Reply:
I think that's clear with "Avoid re-exporting them from a generic `index` file", i don't think we need to add everywhere we're trying to remove this XYZ, it's more implied imo?